src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,426 @@
+    def _calculate_charge_stats(
+        self,
+        charges,
+        broken_pixels,
+        trigger_times,
+    ):
+        """
+        Calculate the sample statistics regarding the charge of each camera pixel.
+
+        Parameters
+        ----------
+        charges: list
+        broken_pixels: list
+        trigger_times: list
+
+        Returns
+        -------
+        charge_stats : dict
+        The sample statistics of the charge of each camera pixel.
+        """
+
+        # ensure numpy array
+        masked_charges = np.ma.array(charges, mask=broken_pixels)
+
+        # mean and std over the sample per pixel
+        max_sigma = self.sigma_clipping_max_sigma
+        pixel_mean, pixel_median, pixel_std = sigma_clipped_stats(
+            masked_charges,
+            sigma=max_sigma,
+            maxiters=self.sigma_clipping_iterations,
+            cenfunc="mean",
+            axis=0,
+        )
+
+        # mask pixels without defined statistical values (mainly due to hardware problems)
+        pixel_mean = np.ma.array(pixel_mean, mask=np.isnan(pixel_mean))
+        pixel_median = np.ma.array(pixel_median, mask=np.isnan(pixel_median))
+        pixel_std = np.ma.array(pixel_std, mask=np.isnan(pixel_std))
+
+        unused_values = np.abs(masked_charges - pixel_mean) > (max_sigma * pixel_std)
+
+        # only warn for values discard in the sigma clipping, not those from before
+        outliers = unused_values & (~masked_charges.mask)
+
+        # add outliers identified by sigma clipping for following operations
+        masked_charges.mask |= unused_values
+
+        # median of the median over the camera
+        median_of_pixel_median = np.ma.median(pixel_median, axis=1)
+
+        # median of the std over the camera
+        median_of_pixel_std = np.ma.median(pixel_std, axis=1)
+
+        # std of the std over camera
+        std_of_pixel_std = np.ma.std(pixel_std, axis=1)
+
+        # outliers from median
+        charge_deviation = pixel_median - median_of_pixel_median[:, np.newaxis]
+
+        charge_median_outliers = np.logical_or(
+            charge_deviation
+            < self.charge_median_cut_outliers[0]
+            * median_of_pixel_median[:, np.newaxis],
+            charge_deviation
+            > self.charge_median_cut_outliers[1]
+            * median_of_pixel_median[:, np.newaxis],
+        )
+
+        # outliers from standard deviation
+        deviation = pixel_std - median_of_pixel_std[:, np.newaxis]
+        charge_std_outliers = np.logical_or(
+            deviation
+            < self.charge_std_cut_outliers[0] * std_of_pixel_std[:, np.newaxis],
+            deviation
+            > self.charge_std_cut_outliers[1] * std_of_pixel_std[:, np.newaxis],
+        )
+
+        return {
+            "sample_time": (
+                trigger_times[0] + (trigger_times[-1] - trigger_times[0]) / 2
+            ).unix
+            * u.s,
```

Comment:
I think you need at least one of sample_time_{min, max} as timestamps to match the current event (trigger_times) with the related set of calibration coefficients.